struct.js

```diff
@@ -21,12 +21,18 @@
 module.exports = StructRW;
 
 var inherits = require('util').inherits;
+var TypedError = require('error/typed');
 
 var LengthResult = require('./base').LengthResult;
 var WriteResult = require('./base').WriteResult;
 var ReadResult = require('./base').ReadResult;
 var BufferRW = require('./base').BufferRW;
 
+var MissingFieldError = TypedError({
+    type: 'bufrw.missing.field',
```

Comment:
this is the first typed error to start namespacing, I'd rather you leave it out in this pr and we namespace them all in one go